identity_did/src/did_url.rs

```diff
@@ -535,6 +534,33 @@ pub(crate) const fn is_char_fragment(ch: char) -> bool {
   is_char_path(ch) || ch == '?'
 }
 
+pub(crate) fn is_valid_percent_encoded_char(s: &str) -> bool {
+  let mut chars = s.chars();
+  let Some('%') = chars.next() else { return false };
+  chars.take(2).all(|c| c.is_ascii_hexdigit() && !c.is_ascii_lowercase())
```

Comment:
Hm, is there a reason, that we disallow lowercase hex encoding here?

> The uppercase hexadecimal digits 'A' through 'F' are equivalent to
   the lowercase digits 'a' through 'f', respectively.  If two URIs
   differ only in the case of hexadecimal digits used in percent-encoded
   octets, they are equivalent.  For consistency, URI producers and
   normalizers should use uppercase hexadecimal digits for all percent-
   encodings.

(from [RFC](https://www.rfc-editor.org/rfc/rfc3986#section-2.1))

Quick check in browser also considers them equal:

```js
decodeURIComponent('%3b') === decodeURIComponent('%3B')
> true
```